style/git_and_github.qmd

```diff
@@ -1,15 +1,52 @@
-# Using git and GitHub
+# Using Git and GitHub
 
 * Commits should be [atomic](https://dev.to/samuelfaure/how-atomic-git-commits-dramatically-increased-my-productivity-and-will-increase-yours-too-4a84)
-* All code should be committed via pull request to main. **Never** push to main
+* All code should be committed via pull request to main. **Never** push to main.
 
 Workflow for writing code with Git and GitHub:
 
-* If you haven't already file an issue that describes what you are doing- whether it be fixing a bug, adding a feature, or something else
-* Make a new branch to work on that feature. GitHub offers a button to do this automatically on the issue page, on the right hand side (see image)
+* If you haven't already, file an issue that describes what you are doing- whether it be fixing a bug, adding a feature,
+  or something else
+* Make a new branch to work on that feature. GitHub offers a button to do this automatically on the issue page, on the
+  right-hand side (see image)
 * Work on the branch, pushing your code regularly to GitHub so it can be run and inspected when you are not around
-* When it's ready merge the branch to main. If you used the create branch button on GitHUb this will automatically close the issue. If the issue is not automatically closed, then close it manually
+* When it's ready merge the branch to main. If you used the create branch button on GitHub this will automatically close
+  the issue. If the issue is not automatically closed, then close it manually
 
+## Review workflow
+
+When you have created a pull request (PR) and marked it ready for review, you must do two things:
+
+1. immediately make someone an assignee - this is the person who will merge the pull request. Typically, this should be
+   the person creating the pull request, i.e. yourself.
+2. ask for reviews from others.
+
+Wait until the reviewers have completed their reviews and marked them as ready to merge. At this point, the person who
+is assigned to the pull request can complete the merge.
+
+By using this approach it ensures that code quality is maintained via requests, and it prevents code from being merged
+when it is not yet ready. For example, you may have started a pull request thinking your work is ready. A reviewer
+checks your code and agrees that it is ready to merge. However, there are occasions where you may realise that there are
+still things to work on before merging that PR, or there are issues that need to be addressed.
+
+The person who is assigned to the PR should be the only person making commits to the branch. This will prevent merge
+conflicts within the branch that is being worked on. If you wish for someone else to collaborate on the branch, then
+you should assign the PR to that person. At that point, they can pull your branch down and work on top of it, but you
+must stop using that branch locally (unless you are reassigned to the branch, at which point you should immediately pull
```

Comment:
Might be clearer to replace from `At that point, ....` to something like:

- Only one person should ever work on changes to a branch at any time,
- It is good practice to always pull changes when a PR is assigned to you.

Reply:
The latter of those is sort of missing the point - if you are assigned then you must be the most up to date version. I prefer the first option

Reply:
I just was getting muddled with the progression of logic in the sentences as it's covering a series of events: PR from person A assigned to person B then back to person A (only then requires a pull to update?). But if person B has that branch locally on the first passing it's possible they haven't updated it for a while so a pull would be good to do?

Reply:
I've left this statement as is, but added a callout to highlight that only one person should be working on a branch